File: ios/SpentyAI/SpentyAI/Core/Components/CurrencyText.swift
import SwiftUI

struct CurrencyText: View {
    let amount: Double
    var currencyCode: String = "INR"
    var font: Font = SpentyFonts.body
    var color: Color?

    private var formattedAmount: String {
        let formatter = NumberFormatter()
        formatter.numberStyle = .currency
        formatter.currencyCode = currencyCode
        formatter.maximumFractionDigits = 2
        formatter.minimumFractionDigits = 2
        return formatter.string(from: NSNumber(value: amount)) ?? "\(amount)"
    }

    private var displayColor: Color {
        if let color { return color }
        return amount >= 0 ? .spentySuccess : .spentyError
    }

    var body: some View {
        Text(formattedAmount)
            .font(font)
            .foregroundColor(displayColor)
    }
}

#Preview {
    VStack {
        CurrencyText(amount: 1234.56)
        CurrencyText(amount: -789.00, currencyCode: "USD")
    }
}

File: android-native/app/src/main/java/com/spentyai/app/core/components/CurrencyText.kt
package com.spentyai.app.core.components

import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import com.spentyai.app.core.theme.SpentyError
import com.spentyai.app.core.theme.SpentySuccess
import com.spentyai.app.core.theme.SpentyType
import java.text.NumberFormat
import java.util.Currency
import java.util.Locale

enum class CurrencySize {
    LARGE, MEDIUM, SMALL
}

@Composable
fun CurrencyText(
    amount: Double,
    currencyCode: String = "USD",
    size: CurrencySize = CurrencySize.SMALL,
    colorBySign: Boolean = false,
    modifier: Modifier = Modifier,
    overrideColor: Color? = null
) {
    val textStyle: TextStyle = when (size) {
        CurrencySize.LARGE -> SpentyType.AmountLarge
        CurrencySize.MEDIUM -> SpentyType.AmountMedium
        CurrencySize.SMALL -> SpentyType.AmountSmall
    }

    val color = overrideColor ?: when {
        colorBySign && amount > 0 -> SpentySuccess
        colorBySign && amount < 0 -> SpentyError
        else -> MaterialTheme.colorScheme.onSurface
    }

    val formatted = formatCurrency(amount, currencyCode)

    Text(
        text = formatted,
        style = textStyle,
        color = color,
        modifier = modifier
    )
}

fun formatCurrency(amount: Double, currencyCode: String = "USD"): String {
    return try {
        val format = NumberFormat.getCurrencyInstance(Locale.US)
        format.currency = Currency.getInstance(currencyCode)
        format.maximumFractionDigits = 2
        format.minimumFractionDigits = 2
        format.format(amount)
    } catch (e: Exception) {
        "$${String.format("%.2f", amount)}"
    }
}
